frontend/public/components/cluster-settings/related-objects.tsx

```diff
@@ -0,0 +1,89 @@
+import * as React from 'react';
+import * as classNames from 'classnames';
+import * as _ from 'lodash-es';
+import { sortable } from '@patternfly/react-table';
+import { Table, TableRow, TableData, RowFunctionArgs } from '../factory';
+import {
+  referenceForModel,
+  ClusterOperator,
+  ClusterOperatorObjectReference,
+  useModelFinder,
+} from '../../module/k8s';
+import { ResourceLink } from '../utils';
+
+const tableColumnClasses = [
+  '',
+  classNames('pf-m-hidden', 'pf-m-visible-on-sm'),
+  classNames('pf-m-hidden', 'pf-m-visible-on-sm'),
+];
+
+const Header = () => [
+  {
+    title: 'Name',
+    sortField: 'name',
+    transforms: [sortable],
+    props: { className: tableColumnClasses[0] },
+  },
+  {
+    title: 'Kind',
+    sortField: 'resource',
+    transforms: [sortable],
+    props: { className: tableColumnClasses[1] },
+  },
+  {
+    title: 'Namespace',
+    sortField: 'namespace',
+    transforms: [sortable],
+    props: { className: tableColumnClasses[2] },
+  },
+];
+
+const Row: React.FC<RowFunctionArgs> = ({ obj, index, key, style, customData: { findModel } }) => {
+  const { name, resource, namespace, group } = obj;
+  const id = `${index}-${key}`;
+  const model = findModel(group, resource);
+
+  const gsv = model ? referenceForModel(model) : '';
+  return (
+    <TableRow id={id} index={index} trKey={key} style={style}>
+      <TableData className={tableColumnClasses[0]}>
+        <ResourceLink kind={gsv} name={name} namespace={namespace} />
+      </TableData>
+      <TableData className={tableColumnClasses[1]}>{resource}</TableData>
+      <TableData className={tableColumnClasses[2]}>
+        {!_.isEmpty(namespace) ? <ResourceLink kind="Namespace" name={namespace} /> : ''}
+      </TableData>
+    </TableRow>
+  );
+};
+
+const RelatedObjects: React.FC<RelatedObjectsProps> = (props) => {
+  const { findModel } = useModelFinder();
+  return (
+    <div className="co-m-pane__body">
+      <Table
+        {...props}
+        Header={Header}
+        Row={Row}
+        customData={{ findModel }}
+        aria-label="Related Objects"
+      />
+    </div>
+  );
+};
+
+const RelatedObjectsPage: React.FC<RelatedObjectsPageProps> = (props) => {
+  const relatedObject: ClusterOperatorObjectReference[] = props.obj?.status?.relatedObjects;
+  const data = relatedObject.filter(({ name, resource }) => name && resource);
```

Comment:
If you use optional chaining on the line above above, you need it here, too.

```suggestion
  const data = relatedObject?.filter(({ name, resource }) => name && resource);
```